src/js/components/RadioButton/RadioButton.js

```diff
@@ -81,9 +81,7 @@ const RadioButton = forwardRef(
       >
         <StyledRadioButton
           flex={false}
-          margin={
-            label ? { right: theme.radioButton.gap || 'small' } : undefined
-          }
+          margin={label ? theme.radioButton.label?.margin : undefined}
```

Comment:
Changing this to move the "right" into base.js seems like it could have breaking changes for themes that just defined a string (expecting it to behave just on the right) and we're also changing an existing theme object name.

Feels like we should keep as is with "gap" for backwards compatibility and apply it on the right as it is.

Thoughs? @jcfilben @britt6612